dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -485,40 +463,56 @@ public CloseableClient build() {
 
             InetSocketAddress socksProxyAddress = getSocksProxyAddress(conf);
             SSLSocketFactory rawSocketFactory = conf.sslSocketFactory();
-            Supplier<Socket> simpleSocketCreator = socksProxyAddress == null
-                    ? () -> new Socket(Proxy.NO_PROXY)
-                    : () -> new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
+            SSLSocketFactory instrumentedSocketFactory =
+                    MetricRegistries.instrument(conf.taggedMetricRegistry(), rawSocketFactory, name);
+            DetachedSocketFactory plainSocketFactory = new SocksSupportingDetachedSocketFactory(socksProxyAddress);
+
+            TlsSocketStrategy tlsStrategy = new DialogueTlsSocketStrategy(
+                    instrumentedSocketFactory,
+                    TlsProtocols.get(),
+                    supportedCipherSuites(CipherSuites.allCipherSuites(), rawSocketFactory, name),
+                    new InstrumentedHostnameVerifier(new DefaultHostnameVerifier(), name, conf.taggedMetricRegistry()));
 
             ConnectInstrumentation connectInstrumentation =
                     new ConnectInstrumentation(conf.taggedMetricRegistry(), name);
 
+            InstrumentedDnsResolver instrumentedDnsResolver = new InstrumentedDnsResolver(
+                    SystemDefaultDnsResolver.INSTANCE, dnsResolver, name, conf.taggedMetricRegistry());
+
+            HttpClientConnectionOperator operator =
+                    new DefaultHttpClientConnectionOperator(
+                            plainSocketFactory,
+                            null,
+                            instrumentedDnsResolver,
+                            RegistryBuilder.<TlsSocketStrategy>create()
+                                    .register(URIScheme.HTTPS.id, tlsStrategy)
+                                    .build()) {
+                        private static final String CONNECT_BEGAN_ATTRIBUTE = "onBeforeSocketConnectNanoTime";
+
+                        @Override
+                        protected void onBeforeSocketConnect(HttpContext httpContext, HttpHost endpointHost) {
+                            super.onBeforeSocketConnect(httpContext, endpointHost);
+                            httpContext.setAttribute(CONNECT_BEGAN_ATTRIBUTE, System.nanoTime());
```

Comment:
I don't love boxing this `long`->`Long`.
The `onBefore`/`onAfter` methods don't allow us to collect timing info when requests fail either, since `onAfterSocketConnect` is never called in that case